assembly-h2o/build.gradle

```diff
@@ -0,0 +1,172 @@
+sparkling water from our repository if it hasn't been already downloaded.
+If this is run without any argument, then basic h2o jar is downloaded,
+but this command takes also argument specifying hadoop version, such as cdh5.4. In case the argument is specified, instead
+of downloading h2o jar this command will download h2o driver jar for specified hadoop version.
+
+Gradle property has higher priority if both H2O_ORIGINAL_JAR and -PdownloadAndExtend are set
+
+Few examples:
+
+1)
+export H2O_ORIGINAL_JAR = ...
+./gradlew build -x check
+
+In this case the jar to be extended is located using the provided environment variable
+
+2)
+./gradlew build -x check -PdownloadAndExtend
+
+In this case the jar to be extended is h2o jar and is downloaded from our repository first
+
+3)
+./gradlew build -x check -PdownloadAndExtend=cdh5.4
+
+In this case the jar to be extended is h2o driver jar for provided hadoop version and is downloaded from our repository first
+
+4)
+./gradlew build -x check -PdownloadAndExtend=abc
+
+This case will throw an exception since such hadoop version is not supported
+
+5)
+export H2O_ORIGINAL_JAR = ...
+./gradlew build -x check -PdownloadAndExtend
+
+This version will ignore environment variable and jar to be extended will be downloaded from our repository. The same
+holds for version with argument.
+"""
+
+configurations {
+    artifactsToMerge
+}
+
+def originalJar = getOrDownloadOriginalJar()
+
+dependencies {
+    artifactsToMerge("ai.h2o:h2o-scala_${scalaBaseVersion}:${h2oVersion}") { transitive = false }
+    artifactsToMerge("org.scala-lang:scala-library:${scalaVersion}")
+    originalJar = getOrDownloadOriginalJar()
+    if (originalJar != null) {
+        artifactsToMerge files(originalJar)
+    }
+}
+
+task extendJar(type: ShadowJar) {
+    configurations = [project.configurations.artifactsToMerge]
+    if (originalJar != null) {
+        def h2oJarFile = new File(originalJar).getName().toLowerCase()
+        if(h2oJarFile.contains("h2odriver")){
+            baseName = "h2odriver_extended"
+        }else{
+            baseName = "h2o_extended"
+        }
+    } else {
+        baseName = null
+    }
+    classifier = null
+    version = null
+}
+
+if (originalJar != null) {
+    build.dependsOn extendJar
+}
+
+/**
+ * Method determining whether the h2o or h2o driver jar is available.
+ * Location to the jar is returned if it's available, oterwise null is returned
+ *
+ * If the user specified Gradle argument downloadAndExtend then the method will try do download the jar and  return path to it
+ */
+def String getOrDownloadOriginalJar() {
+    if (project.hasProperty("downloadAndExtend")) {
+        def arg = project.downloadAndExtend
+        if (arg == null || arg.isEmpty()) {
+            def url = "http://h2o-release.s3.amazonaws.com/h2o/${h2oMajorName != "master" ? "rel-${h2oMajorName}" : "master"}/${h2oBuild}/h2o-${h2oMajorVersion}.${h2oBuild}.zip"
```

Comment:
I was looking into S3 storage: you do not need to download the `zip` file. use this one:

https://s3.amazonaws.com/h2o-release/h2o/rel-turin/4/Rjar/h2o.jar